src/view/pane/message.js

```diff
@@ -110,8 +110,20 @@ Candy.View.Pane = (function(self, $) {
         timestamp = Candy.Util.iso8601toDate(timestamp);
       }
 
+      var messagePane = self.Room.getPane(roomJid, ".message-pane");
```

Comment:
We cleanly handle the case where a message does not have an ID, but there's no way to implement this for clients which generate non-unique IDs. I'll post some more cogent thoughts once I'm really awake.